### ISSUE TITLE
Word reverse in bread crumbs && autocomplete window 

### ISSUE THREAD

**Issue by goget-milk** [AstroNvim/AstroNvim#1358]
### Checklist

- [X] Using a stable version of Neovim (i.e. not neovim nightly)
- [X] `:PackerSync`
- [X] `:AstroUpdate`
- [X] Restarted AstroNvim

### AstroNvim version

v2.7.4

### Neovim version (>= 0.8)

NVIM v0.8.0

### Operating system/version

ArchLinux

### Describe the bug

I noticed the words in reverse order
So far only in .go .js .ts files
In the settings, only the color theme has been changed.
I have two questions:
  1. How to fix the reverse?
  2. How to swap entries in the drop down menu like this...

![like this](https://user-images.githubusercontent.com/7228095/173782308-d28623d9-078d-4d15-a5e1-075899fb41ab.png)

**P.S.** Reinstall AstroNvim. Problem still exists

### Steps to Reproduce

-

### Expected behavior

-

### Screenshots

[![revers.png](https://i.postimg.cc/6pgDmLJc/revers.png)](https://postimg.cc/fV723dbS)

[![revers2.png](https://i.postimg.cc/nr3WYqGv/revers2.png)](https://postimg.cc/sGZK3Q8X)

**Comment by mehalter** [comment]
@goget-milk I can't seem to replicate this with a base installation of AstroNvim. Can you replicate this without any user configuration?

**Comment by mehalter** [comment]
Also for your 2nd question: https://astronvim.github.io/Recipes/cmp#customize-source-priority

here is the relevant documentation

**Comment by goget-milk** [comment]
@mehalter 
Yes i can reproduce
I just reinstall AstroNvim and get this bug on a clean install.
> P.S. Reinstall AstroNvim. Problem still exists

**Comment by mehalter** [comment]
Can you please describe exact steps to reproduce?

**Comment by goget-milk** [comment]
1. Clean install AstroNvim
2. Mason Install: see screenshot 
[![mason.png](https://i.postimg.cc/02yXbyJR/mason.png)](https://postimg.cc/PN0Q7h42)
4. TSInstall: bash css scss dockerfile go gomod gowork html javascript json lua markdown python sql toml typescript vim yaml
5. Install colorschemes: see screenshot
[![plugins.png](https://i.postimg.cc/Wzb8F5z6/plugins.png)](https://postimg.cc/KkwBCrnR)
7. Some keymap: see screenshot
[![keymap.png](https://i.postimg.cc/YqPR0fHY/keymap.png)](https://postimg.cc/vcWfSnDZ)
It's all...

**Comment by mehalter** [comment]
What filetypes are you experiencing this in and what language servers are shown to be attached?

**Comment by goget-milk** [comment]
language servers: bash css dockerfile emmet-ls gopls html json lua prettier pyright python sqls svelte typescript vim yaml
It's all in the screenshot above...
Filetypes: .go .js .ts .svelte

**Comment by mehalter** [comment]
I see that those are the ones installed. Which ones are running in the files that you see these reversing issues? Are you seeing them in all file types? You originally said just `go` files. When you open a go file for example, what language servers are running

**Comment by mehalter** [comment]
Also you mention you can replicate the bug with a clean installation. Can you give specific steps to get to the bug without any user configuration changes? Maybe an example file to open, just installing the bare minimum language servers from Mason, etc.

**Comment by goget-milk** [comment]
in .go files - gopls and treesitter

**Comment by mehalter** [comment]
Thanks, could you please provide the same for `js` and `ts`

**Comment by goget-milk** [comment]
in .svelte - only svelte language server
in .ts .js - tsserver, prettier, treesitter

**Comment by mehalter** [comment]
Could you please provide some replication steps to get to the bug you are describing?

**Comment by goget-milk** [comment]
How did I do...
Clean install of AstroNvim
Installing language servers (Mason install)
Installing treesitter for these languages (TSInstall)
And it's all!

**Comment by mehalter** [comment]
Could you please provide maybe an example file? It looks like some LSP entries aren't reversed? What symbols types are reversed, etc.? Just some details that can get me to see the error in some way on my machine.

**Comment by mehalter** [comment]
For example in your first screenshot the breadcrumbs are not backwards but the word field is, in the second screenshot the breadcrumbs are backwards and structure is backwards, but module isn't backwards. Do you have clear steps that can replicate the bug (assuming I have the language servers set up)

1. `nvim test.go`
2. Type ...
3. See bug in cmp menu if you type `...`
4. Breadcrumbs flip when X happens

**Comment by mehalter** [comment]
Also do you get this bug in all colorschemes or just one or some?

**Comment by mehalter** [comment]
Which colorscheme are you using in the screenshots you sent?

**Comment by goget-milk** [comment]
For example...
svelte - instead of Field -> dleiF
go - instead of Struct -> tcurtS

**Comment by goget-milk** [comment]
> Also do you get this bug in all colorschemes or just one or some?    

This bug in all colorschemes, and also on default. 
I tried without colorschemes
> Which colorscheme are you using in the screenshots you sent?   

I am currently using substrata color
The color scheme does not affect this bug, I checked first

> Do you have clear steps that can replicate the bug (assuming I have the language servers set up)
i need 5 minutes

**Comment by goget-milk** [comment]
nvim main.go -> 
```go
package main

type Book struct {
  ID string `json:"id"`
  Name string `json:"name"`
  Author string `json:"author"`
}
```
This enough to reverse breadcrumbs... when cursor is in structure

**Comment by goget-milk** [comment]
[![structure.png](https://i.postimg.cc/KzGQRtG2/structure.png)](https://postimg.cc/7CRSRGJs)

**Comment by goget-milk** [comment]
[![structure.png](https://i.postimg.cc/RVg3Bf8t/structure.png)](https://postimg.cc/rKtyGdqy)

**Comment by mehalter** [comment]
![2022-11-21_07:28:41_screenshot](https://user-images.githubusercontent.com/1591837/203055422-a9750467-4725-4fc5-a732-d1e81c1440b7.png)

![2022-11-21_07:28:36_screenshot](https://user-images.githubusercontent.com/1591837/203055458-8e9e0897-8f17-43dc-a291-4c208143609d.png)

Unable to replicate this behavior. Not sure what could be causing this

**Comment by goget-milk** [comment]
Can the old config or cache or temp files affect this behavior?
Where can I clean the system from old neovim files?

**Comment by mehalter** [comment]
I don't believe the config or cache would have any affect. I honestly have no idea how some words could even be reversed, this isn't part of any of our logic at all. The easiest way to clean the old neovim files is:

```sh
rm -rf ~/.local/share/nvim
rm -rf ~/.local/state/nvim
rm -rf ~/.cache/nvim
nvim
```

Also might be worth checking if you can replicate this in other terminals as well

**Comment by goget-milk** [comment]
>Also might be worth checking if you can replicate this in other terminals as well

You are right again!
Everything is fine in another terminal (kitty), default (alacritty)
Why didn't I think to check this earlier?
Although LunarVim worked fine.....
Thank you anyway

**Comment by goget-milk** [comment]
P.S. Is it possible to swap the icon title definition in places in the auto-completion drop-down list?

**Comment by mehalter** [comment]
yeah, you will just need to customize the `formatting.format` function in your `plugins.cmp` table. I'm not sure exactly what it should be, but it will just take some amount of lua to get it working correctly. You can check out the `cmp` docs for information on customizations: https://github.com/hrsh7th/nvim-cmp/wiki/Menu-Appearance#basic-customisations

You'll just want to make sure you use `lspkind` to get the icons and look into their API to make sure you make a custom format function correctly.

**Comment by cadenhenrich** [comment]
@goget-milk I have tested alacritty and it seems to work fine, just as it does in kitty. I have found, however, that in the distribution of alacritty with ligature support the text is reversed, and also seems to be printing right-to-left (as if [this](https://unicode-explorer.com/c/202E) is being inserted somewhere). Hope this helps for anyone still experiencing the issue, I myself have resigned to using the normal version of alacritty with no ligatures.

Cheers :)
